File: src/main/java/gg/uhc/migration/configuration/LongListReader.java
package gg.uhc.migration.configuration;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.configuration.InvalidConfigurationException;

import java.util.List;

public class LongListReader extends ConfigurationReader<List<Long>> {
    @Override
    protected List<Long> read(ConfigurationSection section, String key) throws InvalidConfigurationException {
        if (!section.isList(key)) throw up(WRONG_TYPE, key, "list of numbers", section.get(key));

        return section.getLongList(key);
    }
}

File: src/main/java/gg/uhc/migration/configuration/LongListReader.kt
package gg.uhc.migration.configuration

import org.bukkit.configuration.ConfigurationSection

fun ConfigurationSection.toLongList(key: String): List<Long> {
    if (!this.isList(key))
        throw wrongType(key, "list of numbers", this.get(key))

    return this.getLongList(key)
}